File: src/main/java/land/leets/domain/portfolio/presentation/dto/PortfoliosResponse.java
package land.leets.domain.portfolio.presentation.dto;

import jakarta.validation.constraints.NotBlank;
import land.leets.domain.portfolio.domain.Portfolio;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class PortfoliosResponse {

    @NotBlank
    private Long portfolioId;

    @NotBlank
    private String name;

    @NotBlank
    private String mainImgName;

    public static PortfoliosResponse from(Portfolio portfolio) {
        return new PortfoliosResponse(
                portfolio.getPortfolioId(),
                portfolio.getName(),
                portfolio.getMainImgName()
        );
    }
}

File: src/main/kotlin/land/leets/domain/portfolio/presentation/dto/PortfoliosResponse.kt
package land.leets.domain.portfolio.presentation.dto

import land.leets.domain.portfolio.domain.Portfolio

data class PortfoliosResponse(
    val portfolioId: Long,
    val name: String,
    val mainImgName: String,
) {
    companion object {
        fun from(portfolio: Portfolio): PortfoliosResponse {
            return PortfoliosResponse(
                portfolioId = portfolio.portfolioId,
                name = portfolio.name,
                mainImgName = portfolio.mainImgName
            )
        }
    }
}
